registry/storage/driver/storagedriver.go

```diff
@@ -106,11 +106,11 @@ type FileWriter interface {
 }
 
 // PathRegexp is the regular expression which each file path must match. A
-// file path is absolute, beginning with a path components and may contain
-// more of them separated by slashes, where each path component is
+// file path is absolute, beginning with a slash or a path components and may
+// contain more of them separated by slashes, where each path component is
 // restricted to alphanumeric characters or a period, underscore, colon or
 // hyphen.
-var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]+(/[A-Za-z0-9._:-]+)*)+$`)
+var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]*(/[A-Za-z0-9._:-]+)*)+$`)
```

Comment:
So, this changes the meaning of a valid path in an odd way. I'm not sure this is quite correct. Typically, there is always a backend root, and paths are accessed as subpaths of those roots. For example, we have the following:

```
asdf/qwer
```

This is then combined with whatever the backend has, which might be a slash or a root path or a url. The problem with allowing an extra slash is following:

```
join(/backend, asdf/qwer) -> /backend/asdf/qwer)
join(/backend, /asdf/qwer) -> /backend//asdf/qwer)
```

These two paths may not be the same, depending on the backend.

I _think_ we have a fix that accepts and removes a leading slash, but you'll have to hunt it down. I am not sure which version your currently running on, so it is hard to tell if this is a fix.

I know this seems like an innocent change, but it has ramifications.